File: sdk/python/costorah/instrumentation/azure_openai.py
"""
AzureOpenAIInstrumentor — automatic usage capture for Azure OpenAI
Service calls made through the official `openai` package's
`AzureOpenAI`/`AsyncAzureOpenAI` clients.

    from openai import AzureOpenAI
    from costorah.instrumentation import AzureOpenAIInstrumentor

    AzureOpenAIInstrumentor().instrument()

    client = AzureOpenAI(azure_endpoint=..., api_version=...)
    client.chat.completions.create(model="my-deployment", messages=[...])

Shares the same patched `Completions`/`Responses` classes as every other
OpenAI-family instrumentor (see `_openai_compatible.py`); only Azure
clients (detected by class name) are captured while this instrumentor is
active.
"""

from __future__ import annotations

from costorah.instrumentation._openai_compatible import _OpenAICompatibleInstrumentor


class AzureOpenAIInstrumentor(_OpenAICompatibleInstrumentor):
    name = "azure_openai"
    fixed_provider = "azure_openai"

File: sdk/javascript/src/instrumentation/azureOpenai.ts
/**
 * AzureOpenAIInstrumentor — automatic usage capture for Azure OpenAI
 * Service calls made through the official `openai` package's
 * `AzureOpenAI` client. Shares the same patched Completions/Responses
 * prototypes as every other OpenAI-family instrumentor (see
 * openaiCompatible.ts); only Azure clients (detected by class name) are
 * captured while this instrumentor is active.
 */

import { OpenAICompatibleInstrumentor } from "./openaiCompatible.js";

export class AzureOpenAIInstrumentor extends OpenAICompatibleInstrumentor {
  readonly name = "azure_openai";
  readonly fixedProvider = "azure_openai";
}
